### PR TITLE
Kennric/test data script

### PR DESCRIPTION
**Pull request by Kennric**
fixes issue #253 

## Changes in this PR.

- [X] Data export method
- [X] Data import method
- [X] Data anonymization methods
- [X] Rake tasks for import/export
- [X] Docs

## Testing this PR.

1. On staging: `rvm all do rake export_data`
2. Examine the generated data to make sure it is anonymized
3. Copy data to local instance (or copy from fir:/tmp/test_data)
4. `docker-compose run dev bash`
5. `rake import_data`
6. exit and `docker-compose up dev`
7. visit localhost:4567 and view your data in the app

@osuosl/devs

@alxngyn @hsolorza @ramereth

### CODE REVIEW

File: docs/source/development/test_data.rst [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  rake export_data
+
+or
+
+.. code-block:: bash
+
+  rvm all do rake export_data
+
+If you would like to set a different time frame, set the EXPORT_DATA_DAYS
+environment variable.
```
Comment (ramereth) [inline docs/source/development/test_data.rst]: Can you give an example of what we would set in ``EXPORT_DATA_DAYS``?

File: docs/source/development/test_data.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+subdirectory. Each model or measurement table will be written to a separate
+file named for the model or table.
+
+It's a good idea to browse through the application you will be exporting from
+and make sure that the clients you are exporting have appropriate data. Avoid
+clients whose projects have very large numbers of resources, as this will export
+a large amount of test data. A good selection would include clients with
+different combinations of resources.
+
+By default this will export all measuremnt data collected in the last 60 days.
```
Comment (hsolorza) [inline docs/source/development/test_data.rst]: 'measurement' is mispelled

File: lib/datasampler.rb [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    # get projects
+    projects_filename = @directory + 'projects.json'
+    projects_json = File.open(projects_filename, &:readline)
+    projects = Project.array_from_json(projects_json,
+                                       fields: Project.columns.map(&:to_s))
+
+    projects.each(&:save)
+  end
+
+  # Loop through our definede resources and import the data for each one
```
Comment (hsolorza) [inline lib/datasampler.rb]: 'defined' is misspelled

File: lib/datasampler.rb [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  def initialize
+    @directory = 'test_data/'
+  end
+
+  # Deletes all the extant data. This does not drop tables, only removes rows
+  # rubocop:disable AbcSize
+  def delete_data
+    # deletes everything
+    Report.plugin_matrix.each do |resource_name, measurements|
+      model = Object.const_get(resource_name + 'Resource').camelcase(:upper)
```
Comment (hsolorza) [inline lib/datasampler.rb]: This causes:
NameError: wrong constant name dbResource
/data/code/lib/datasampler.rb:19:in `const_get'
/data/code/lib/datasampler.rb:19:in `block in delete_data'
/data/code/lib/datasampler.rb:18:in `each'
/data/code/lib/datasampler.rb:18:in `delete_data'
/data/code/lib/datasampler.rb:79:in `import_data'
/data/code/Rakefile:82:in `block in <top (required)>'
Tasks: TOP => import_data 

when running `rake import_data`

**Comment by ramereth** [comment]
@Kennric you should automatically create the test_data directory, otherwise it errors out.

**Comment by Kennric** [comment]
@alxngyn the database wiping is a documentation issue - we need to document that dev instances should use a separate db file from the test instance.

**Comment by ramereth** [comment]
@Kennric remind me the steps I need to do to properly import the data into a docker-compose instance? I'm having a hard time getting the data to show up after I load it and run dev up. If you could provide step by step that would be great.

Thanks-

**Comment by Kennric** [comment]
@ramereth 

To import data in the docker dev instance

Assuming you have data exported as described in docs/source/development/test_data.rst in a directory `test_data` in the app directory:

1. in dockerfiles/app.env change the database url to prevent tests from overwriting the dev database - default is test.sqlite, just change it to something else

```
DB_URL=sqlite://dev.sqlite
```

2. `docker-compose run dev bash`
3. `rake import_data`
4. Answer when it asks if you are sure
5. exit and visit `http://localhost:4567`

**Comment by ramereth** [comment]
@Kennric I'm getting the following error on step 2. Did you miss a step I needed to do?

```
Sequel::DatabaseError: SQLite3::SQLException: no such table: plugins
```

**Pull request merged**